src/Autofac/Core/Registration/IPerScopeRegistrationSource.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Autofac.Core.Registration;
+
+/// <summary>
+/// Indicates that a registration source operates per-scope, and should not provide
+/// registrations for requests from child scopes.
+/// </summary>
+[SuppressMessage("Design", "CA1040:Avoid empty interfaces", Justification = "Need a marker interface of per-scope registration sources")]
```

Comment:
Performance primarily, but also a little because it allows us the future option where `IPerScopeRegistrationSource` _does_ have its own members, so we don't have to switch from attributes.

Also, `is IINterface` checks are naturally NativeAOT friendly, custom attributes less so.

Reply:
I'm good with that, just figured I'd ask.